source/features/unlink-image-auto-links.tsx

```diff
@@ -0,0 +1,46 @@
+import select from 'select-dom';
+import * as pageDetect from 'github-url-detection';
+
+import features from '.';
+import GitHubURL from '../github-helpers/github-url';
+import {isMarkdownFile} from '../github-helpers';
+
+function isMarkupFile(): boolean {
+	if (isMarkdownFile()) {
+		return true;
+	}
+
+	// https://github.com/github/markup/blob/46908504f5099b15a88030749f0aeb52cd2f6799/README.md#markups
+	if (/\.(textile|rdoc|org|creole|(media)?wiki|rst|a(scii)?doc|asc|pod)$/.test(location.pathname)) {
+		return true;
+	}
+
+	return false;
+}
+
+function getRawUrl(link: HTMLAnchorElement): string {
+	if (pageDetect.isSingleFile(link)) {
+		const rawLink = new GitHubURL(link.href).assign({route: 'raw'});
+		return rawLink.href;
+	}
+
+	return link.href;
+}
+
+function init(): void {
+	for (const image of select.all('.markdown-body a > img')) {
+		const link = image.parentElement as HTMLAnchorElement;
+
+		if (image.src === getRawUrl(link)) {
+			link.replaceWith(image);
+		}
+	}
+}
+
+void features.add(__filebasename, {
+	include: [
+		pageDetect.isRepoTree,
+		() => pageDetect.isSingleFile() && isMarkupFile()
```

Comment:
```suggestion
		() => isMarkupFile()
```

`isMarkupFile` should already imply it's single, there should never be a case when `isMarkupFile` is not also a single file